core/src/androidTest/java/org/hisp/dhis/android/core/category/CategoryComboStoreShould.java

```diff
@@ -56,33 +58,60 @@ public void insert_and_delete_a_category_combo() throws Exception {
         thenAssertStoreReturnsNotDeleted();
     }
 
-    private void givenACategoryCombo() {
-        newCategoryCombo = generateCategoryCombo();
+    @Test
+    @MediumTest
+    public void update_a_category_combo() {
+
+        whenInsertNewCategoryCombo();
+
+        whenUpdateACategoryCombo();
+
+        thenAssertThatCategoryComboWasUpdated();
+
     }
 
-    private CategoryCombo generateCategoryCombo(){
-        Date today = new Date();
+    @Test
+    @MediumTest
+    public void delete_all_categories_combos_from_db() {
+
+        whenInsertNewCategoryCombo();
+
+        thenAssertCategoryComboWasInserted();
+
+        whenDeleteAllCategoriesCombosFromDB();
+
+        thenAssertThereAreNotCategoryCombosInDB();
 
-        return CategoryCombo.builder()
-                .uid("m2jTvAj5kkm")
-                .code("BIRTHS")
-                .created(today)
-                .name("Births")
-                .displayName("Births")
-                .isDefault(false)
-                .categories(new ArrayList<Category>())
-                .build();
+    }
+
+    @Test
+    @MediumTest
+    public void query_all_category_combos(){
+
+        whenInsertNewCategoryCombo();
+
+        thenAssertCategoryComboWasInserted();
+
+        thenAssertQueryAllBringCategoryCombosFromDB();
+
+    }
+
+    private void givenACategoryCombo() {
+        newCategoryCombo = generateCategoryCombo(DEFAULT_CATEGORY_COMBO_UID);
     }
 
     private void whenInsertNewCategoryCombo() {
+
```

Comment:
this method has two responsibilities, create a fixture (givens...) and realize actions (whens..), I think that the code would be more readable in tests where is used if split this method in two, one responsibility per method, can you refactor it?

Reply:
Sure!